kernel/src/device/fb.rs

```diff
@@ -325,6 +287,109 @@ impl FbHandle {
         current_userspace!().write_val(arg, &screen_info)?;
         Ok(0)
     }
+
+    /// Handles the [`IoctlCmd::GETCMAP`] ioctl command.
+    /// Takes a [`aster_framebuffer::FbCmap`] structure specifying the range as input and outputs the same structure filled with color palette data.
+    fn handle_get_cmap(&self, arg: usize) -> Result<i32> {
+        let cmap_user: FbCmapUser = current_userspace!().read_val(arg)?;
+
+        if cmap_user.len == 0 {
+            return Ok(0);
+        }
+
+        let cmap = self.framebuffer.cmap().lock();
+
+        // Check bounds
+        let start = cmap_user.start as usize;
+        let len = cmap_user.len as usize;
+
+        if start >= cmap.red.len() || start + len > cmap.red.len() {
+            return_errno_with_message!(Errno::EINVAL, "colormap index out of bounds");
+        }
+
+        // Copy color data to userspace
+        let red_slice = &cmap.red[start..start + len];
+        Self::write_color_maps_to_user(cmap_user.red, red_slice)?;
+
+        let green_slice = &cmap.green[start..start + len];
+        Self::write_color_maps_to_user(cmap_user.green, green_slice)?;
+
+        let blue_slice = &cmap.blue[start..start + len];
+        Self::write_color_maps_to_user(cmap_user.blue, blue_slice)?;
+
+        if let Some(ref transp) = cmap.transp {
+            let transp_slice = &transp[start..start + len];
+            Self::write_color_maps_to_user(cmap_user.transp, transp_slice)?;
+        }
+
+        Ok(0)
+    }
+
+    /// Handles the [`IoctlCmd::PUTCMAP`] ioctl command.
+    /// Takes a [`aster_framebuffer::FbCmap`] structure with color palette data as input and produces no output.
+    fn handle_set_cmap(&self, arg: usize) -> Result<i32> {
+        let cmap_user: FbCmapUser = current_userspace!().read_val(arg)?;
+
+        if cmap_user.len == 0 {
+            return Ok(0);
+        }
+
+        // Size check to prevent excessive memory allocation
+        if cmap_user.len > 256 {
+            return_errno_with_message!(Errno::EINVAL, "colormap too large");
```

Comment:
```suggestion
        // Check the size to prevent excessive memory allocation.
        if cmap_user.len > 256 {
           return_errno_with_message!(Errno::EINVAL, "the color map size is too large");
```